File: iam-jruby/create_user.rb
require 'java'
require 'xlclient'
 
include_class('java.lang.Exception') {|package,name| "J#{name}" }
include_class 'java.lang.System'
include_class 'java.util.HashMap'
include_class 'java.util.Hashtable'
 
 
login = ARGV[0]

jndi = Hashtable.new({
    'java.naming.provider.url' => 't3://adc2101221.us.oracle.com:7001/oim',
    'java.naming.factory.initial' => 'weblogic.jndi.WLInitialContextFactory'
})

xlclient = XLAPIClient.new
xlclient.defaultLogin
#xlclient.passwordLogin('xelsysadm', 'Welcome1')
#xlclient.passwordLoginWithDiscovery('xelsysadm', 'Welcome1', jndi)

usrIntf = xlclient.getIntf('usr')
 
usrHash = {
    'Users.User ID' => login ,
    'Users.First Name' => login + 'First' ,
    'Users.Last Name' => login + 'Last' ,
    'Users.Middle Name' => login + 'Middle' ,
    'Users.Password' => 'Welcome1',
    'Users.Role' => 'Full-Time',
    'Users.Xellerate Type' => 'End-User',
    'Organizations.Key' => '1'
}
 
usrMap = HashMap.new(usrHash)
 
t1 = System.currentTimeMillis
usrKey = usrIntf.createUser(usrMap)
t2 = System.currentTimeMillis
 
delta = t2-t1
puts "Created user with key = #{usrKey} time = #{delta}"

xlclient.logout
System.exit 0

File: iam-scala/create_user.scala
import scala.collection.mutable._
import java.util.HashMap
import java.lang.Long

import com.thortech.xl.util.config.ConfigurationClient
import com.thortech.xl.crypto._
import oracle.iam.platform.OIMClient

import oracle.iam.identity.usermgmt.vo._
import oracle.iam.identity.usermgmt.api._

import OIM._

object CreateUser {
    def main(args: Array[String]) {

        val login: String = args(0)

        // create a user
        val oimClient = getClient("xelsysadm", "Welcome1")
        val usrMgr = getService(oimClient, "usrmgr")

        val usrMap = Map[String,Object](
            "User Login" -> login,
            "First Name" -> (login + "First"),
            "Middle Name" -> (login + "Middle"),
            "Last Name" -> (login + "Last"),
            "usr_password" -> "Welcome1",
            "Role" -> "Full-Time",
            "Xellerate Type" -> "End-User",
            "act_key" -> new Long(1),
            "Common Name" -> login,
            "Email" -> (login + "@oracle.com")
        )

        val u = new User(login, getJavaMap(usrMap))
        val res = usrMgr.create(u)
        val usrKey = res.getEntityId()

        println("User created login = " + login + "  key = " + usrKey)

        oimClient.logout()
    }
}
